### ISSUE TITLE
The result is empty

### ISSUE THREAD

**Issue by huhusmang** [Yunlongs/Goshawk#16]
I successfully configured the environment and ran Goshwak, and then I tried to use openssl as a test. The commit I used was b33c48b75aaf33c93aeda42d7138616b9e6a64cb, an earlier version. I successfully ran it to the end, but the running results were inconsistent with the official sample results, and the number of bugs detected is 0, I don't know if there is really no bug or I made a mistake, here are some results :

This is the content of the AllocCustomizedFile.txt file
```
{"funcname": "ossl_cmp_certrep_new", "returned_object": ["->extraCerts"], "param_object": []}
{"funcname": "dsa_new_intern", "returned_object": ["->lock"], "param_object": []}
{"funcname": "dsa_do_sign_int", "returned_object": ["->r", "->s"], "param_object": []}
{"funcname": "dh_new_intern", "returned_object": ["->lock"], "param_object": []}
{"funcname": "EC_GROUP_new_from_ecparameters", "returned_object": ["->seed"], "param_object": []}
{"funcname": "d2i_ECDSA_SIG", "returned_object": ["->r", "->s"], "param_object": [1, "psig"]}
{"funcname": "rsa_multip_info_new", "returned_object": ["->d", "->pp", "->r", "->t"], "param_object": []}
{"funcname": "policy_data_new", "returned_object": ["->expected_policy_set"], "param_object": []}
{"funcname": "make_IPAddressFamily", "returned_object": ["->ipAddressChoice"], "param_object": []}
{"funcname": "kdf_data_new", "returned_object": ["->lock"], "param_object": []}
{"funcname": "kmac_new", "returned_object": ["->ctx"], "param_object": []}
{"funcname": "ssl_cert_new", "returned_object": ["->lock"], "param_object": []}
{"funcname": "dsa_dupctx", "returned_object": ["->mdctx"], "param_object": []}
{"funcname": "cms_kari_create_ephemeral_key", "returned_object": [], "param_object": [1, "->pctx"]}
{"funcname": "OSSL_CRMF_MSG_create_popo", "returned_object": [], "param_object": [2, "->popo"]}
{"funcname": "dh_key2buf", "returned_object": [], "param_object": [2, "pbuf_out"]}
{"funcname": "hmac_ctx_alloc_mds", "returned_object": [], "param_object": [1, "->i_ctx", 1, "->md_ctx", 1, "->o_ctx"]}
{"funcname": "PKCS12_pbe_crypt", "returned_object": [], "param_object": [6, "data"]}
{"funcname": "pkey_get_rsa", "returned_object": [], "param_object": [2, "rsa"]}
{"funcname": "pkey_get_eckey", "returned_object": [], "param_object": [2, "eckey"]}
{"funcname": "SRP_create_verifier_BN_ex", "returned_object": [], "param_object": [4, "verifier"]}
{"funcname": "X509v3_add_ext", "returned_object": [], "param_object": [1, "x"]}
{"funcname": "X509_add_cert_new", "returned_object": [], "param_object": [1, "sk"]}
{"funcname": "ossl_prov_macctx_load_from_params", "returned_object": [], "param_object": [1, "macctx"]}
{"funcname": "ssl3_cbc_copy_mac", "returned_object": [], "param_object": [4, "mac"]}
{"funcname": "ssl_create_cipher_list", "returned_object": [], "param_object": [3, "cipher_list"]}
```

And this is the some examples of the FreeCustomizedFile.txt file
```
{"funcname": "sk_danetls_record_free", "param_names": [0, "sk"], "member_name": ["sk->data"]}
{"funcname": "sk_danetls_record_pop_free", "param_names": [0, "sk"], "member_name": ["sk->data"]}
{"funcname": "lh_SSL_SESSION_free", "param_names": [0, "lh"], "member_name": ["lh->b"]}
{"funcname": "lh_X509_NAME_free", "param_names": [0, "lh"], "member_name": ["lh->b"]}
{"funcname": "ssl_evp_cipher_free", "param_names": [0, "cipher"], "member_name": ["cipher->lock", "cipher->prov", "cipher->prov->error_strings", "cipher->prov->module", "cipher->prov->module->filename", "cipher->prov->module->loaded_filename", "cipher->prov->module->lock", "cipher->prov->name", "cipher->prov->operation_bits", "cipher->prov->parameters", "cipher->prov->parameters->data", "cipher->prov->path"]}
{"funcname": "sk_EX_CALLBACK_free", "param_names": [0, "sk"], "member_name": ["sk->data"]}
```

And this is the report html:

![image](https://github.com/Yunlongs/Goshawk/assets/77958410/d6aab9f1-a406-4515-b740-1eb2662f5295)

If you need to provide running logs, I have saved them here

**Comment by huhusmang** [comment]
I re-debugged and found that when executing the `CodeChecker log -b "make CC=clang HOSTCC=clang -jN" -o compilation.json` command, an error occurred at the end:
```
/usr/bin/ld: ./libcrypto.so: undefined reference to `v3_utf8_list'
/usr/bin/ld: ./libcrypto.so: undefined reference to `tdes_dinit'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile:24487: test/asn1_decode_test] Error 1
make[1]: Leaving directory '/root/openssl'
make: *** [Makefile:3298: build_sw] Error 2
[INFO 2023-08-17 04:36] - Build failed.
```
so will it have something to do with this

**Comment by Yunlongs** [comment]
@huhusmang The files FreeCustomizedFile.txt and AllocCustomizedFile.txt  are not empty means the MM functions are identified successfully. 

>  and the number of bugs detected is 0
What your means of the number of "bug"?  Is the "bug" means the number of "potential bugs" reported by analyzers?

In OpenSSL, the number of potential bugs should not be zero, i analyzed master branch of OpenSSL several months ago and it was not zero. That means either there is bug existed in CSA checkers or you run with problematic eviorment.

Can you tell what evionment you are used? The Goshawk you used was built by manually or built from Dockerfile or Dockerhub's image?

I have changed Dockerfile last months, maybe that caused some imcompatiabity.

**Comment by huhusmang** [comment]
@Yunlongs Thanks Reply, bro. I just reproduced the project last week. The environment was built with the latest Dockerfile, but there were some problems when building the environment. The following is the Dockerfile after specifying the version of Cython and PyYAML:

``` dockerfile
# syntax=docker/dockerfile:1-labs

FROM ubuntu:22.04

WORKDIR /root

RUN apt update
RUN apt install flex bison libssl-dev libelf-dev bc vim python3 python3-pip lsb-release wget software-properties-common gnupg -y
RUN apt clean 
RUN apt autoremove 
RUN rm -rf /var/lib/apt/lists/* 
RUN pip3 install tensorflow==2.11.0 -i https://pypi.tuna.tsinghua.edu.cn/simple

##### pip new packages to solve the dependency error
RUN pip3 install "Cython<3"
RUN pip3 install PyYAML==5.4.1 --no-cache-dir --no-build-isolation

RUN pip3 install CodeChecker==6.21.0 -i https:pypi.tuna.tsinghua.edu.cn/simple

RUN wget https:apt.llvm.org/llvm.sh \
    && chmod +x llvm.sh \
    && ./llvm.sh 15 \
    && ln -s /usr/bin/clang-15 /usr/bin/clang \
    && ln -s /usr/bin/clang++-15 /usr/bin/clang++

ADD ./ /root/Goshawk/
```

**Comment by Yunlongs** [comment]
@huhusmang When using the new Dockerfile, does your issue of running on OpenSSL still exist?

**Comment by huhusmang** [comment]
@Yunlongs Yes, I'm always running Openssl in the new environment.I have recorded the running log file, it may be a little big. I can email you if I need it.

**Comment by Yunlongs** [comment]
@huhusmang  I reproduced your issue. It happens when GoshawkAnalyzer.so meet imcompatiable version of clang. I will take some days to fix that.

**Comment by Yunlongs** [comment]
Issue reproduced:
```
root@1299f6a9d2c2:~/Goshawk/testsuits# clang -Xclang -analyze -Xclang -load -Xclang ${Goshawk}/plugins/GoshawkAnalyzer.so \
               -Xclang -analyzer-checker=security.GoshawkChecker -Xclang -analyzer-config \
               -Xclang security.GoshawkChecker:MemFuncsDir=${Goshawk}/temp/CSA \
               -Xclang -analyzer-config -Xclang security.GoshawkChecker:PathNumberFile=${Goshawk}/temp/CSA/path_number.txt \
               -Xclang -analyzer-config -Xclang security.GoshawkChecker:ExternFile=${Goshawk}/temp/CSA/extern_count.txt \ 
               -c test.c
warning: checker plugin 'GoshawkAnalyzer.so' is not compatible with this version of the analyzer [-Wanalyzer-incompatible-plugin]
note: current API version is '15.0.7', but plugin was compiled with version '15.0.0'
error: no analyzer checkers or packages are associated with 'security.GoshawkChecker'
note: use -analyzer-disable-all-checks to disable all static analyzer checkers
error: no analyzer checkers or packages are associated with 'security.GoshawkChecker'
error: no analyzer checkers or packages are associated with 'security.GoshawkChecker'
error: no analyzer checkers or packages are associated with 'security.GoshawkChecker'
```

This issue was introduced in https://github.com/Yunlongs/Goshawk/commit/d6af3bd9531e803f3240f8f1698c18c745640d1b.
This commit updated the Dockerfile to change the way of building clang to LLVM's offical build script "llvm.sh".
However, the default clang version of "./llvm.sh 15" is 15.0.7 and it is not the 15.0.0 version that our plugins have built, and the imcompatiablity issue thus occurred.

**Comment by Yunlongs** [comment]
This issue is fixed by #17 .
